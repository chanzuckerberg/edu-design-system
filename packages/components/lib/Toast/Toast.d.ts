import React from "react";
export declare type Color = "success" | "alert";
declare type Props = {
    /**
     * Additional class names passed in for styling.
     */
    className?: string;
    /**
     * The contents of the toast.
     */
    children: React.ReactNode;
    /**
     * The color of the Toast, based on EDS defined colors. Also determines the icon used.
     * Note that the Icon mapping matches the style of Banners.
     */
    color: Color;
    /**
     * Callback when Toast is dismissed.
     */
    onDismiss?: () => void;
};
/**
 * A toast used to provide information on the state of the page, usually in response to a
 * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
 * data was successfully saved.
 */
export default function Toast({ children, className, color, onDismiss, ...rest }: Props): JSX.Element;
export {};
