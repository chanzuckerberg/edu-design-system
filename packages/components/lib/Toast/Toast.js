"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const clsx_1 = __importDefault(require("clsx"));
const react_1 = __importDefault(require("react"));
const CloseButton_1 = __importDefault(require("../CloseButton"));
const Text_1 = __importDefault(require("../Text"));
const Notification_module_css_1 = __importDefault(require("../common/Notifications/Notification.module.css"));
const NotificationIcon_1 = __importDefault(require("../common/Notifications/NotificationIcon"));
const Toast_module_css_1 = __importDefault(require("./Toast.module.css"));
/**
 * A toast used to provide information on the state of the page, usually in response to a
 * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
 * data was successfully saved.
 */
function Toast(_a) {
    var { children, className, color, onDismiss } = _a, 
    // Allow for additional attributes such as aria roles
    rest = __rest(_a, ["children", "className", "color", "onDismiss"]);
    return (react_1.default.createElement("div", Object.assign({ className: (0, clsx_1.default)(className, Toast_module_css_1.default.toastDialog, color === "success" && Notification_module_css_1.default.colorSuccess, color === "alert" && Notification_module_css_1.default.colorAlert) }, rest),
        react_1.default.createElement("div", { className: Toast_module_css_1.default.content },
            react_1.default.createElement(NotificationIcon_1.default, { variant: color }),
            react_1.default.createElement(Text_1.default, { color: "inherit", size: "sm" }, children)),
        onDismiss && react_1.default.createElement(CloseButton_1.default, { color: color, onClose: onDismiss })));
}
exports.default = Toast;
//# sourceMappingURL=Toast.js.map