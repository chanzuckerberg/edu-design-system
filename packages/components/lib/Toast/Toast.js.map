{"version":3,"file":"Toast.js","sourceRoot":"","sources":["../../src/Toast/Toast.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;AAAA,gDAAwB;AACxB,kDAA0B;AAC1B,iEAAyC;AACzC,mDAA2B;AAC3B,8GAA0E;AAC1E,gGAAwE;AACxE,0EAAwC;AAwBxC;;;;GAIG;AACH,SAAwB,KAAK,CAAC,EAOtB;QAPsB,EAC5B,QAAQ,EACR,SAAS,EACT,KAAK,EACL,SAAS,OAGH;IAFN,qDAAqD;IAClD,IAAI,cANqB,+CAO7B,CADQ;IAEP,OAAO,CACL,qDACE,SAAS,EAAE,IAAA,cAAI,EACb,SAAS,EACT,0BAAM,CAAC,WAAW,EAClB,KAAK,KAAK,SAAS,IAAI,iCAAW,CAAC,YAAY,EAC/C,KAAK,KAAK,OAAO,IAAI,iCAAW,CAAC,UAAU,CAC5C,IACG,IAAI;QAER,uCAAK,SAAS,EAAE,0BAAM,CAAC,OAAO;YAC5B,8BAAC,0BAAgB,IAAC,OAAO,EAAE,KAAK,GAAI;YACpC,8BAAC,cAAI,IAAC,KAAK,EAAC,SAAS,EAAC,IAAI,EAAC,IAAI,IAC5B,QAAQ,CACJ,CACH;QACL,SAAS,IAAI,8BAAC,qBAAW,IAAC,KAAK,EAAE,KAAK,EAAE,OAAO,EAAE,SAAS,GAAI,CAC3D,CACP,CAAC;AACJ,CAAC;AA3BD,wBA2BC"}