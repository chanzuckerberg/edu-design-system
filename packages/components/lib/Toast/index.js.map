{"version":3,"file":"index.js","sourceRoot":"","sources":["../../src/Toast/index.ts"],"names":[],"mappings":";;;;;;AAAA,iCAAkC;AAAzB,iHAAA,OAAO,OAAA"}