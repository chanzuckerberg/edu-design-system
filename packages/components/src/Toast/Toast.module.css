.toastDialog {
  @apply w-full max-w-100 pl-4 pr-2 grid gap-2 items-center;
  @apply border border-l-4 rounded shadow-1;

  grid-template-columns: 1fr min-content;
  --svg-icon-size-default: 1.25rem;
}

.content {
  @apply py-4 grid gap-4;

  grid-template-columns: min-content 1fr;
}

/* TODO: consolidate dismiss button styles with the Banner styles */
.dismiss {
  @apply w-12 h-12 justify-center rounded-full;

  /* TODO: test motion-safe with new Clickable motion-reduce change */
  @apply hover:text-white motion-safe:transition ease-out;
}

.dismiss:hover {
  background-color: var(--dismiss-hover-color);
}

/* TODO: consolidate color styles with the Banner styles */
.colorSuccess {
  @apply text-success-700 bg-success-100;

  /* Left border is thicker and a darker color than the surrounding border */
  border-color: var(--eds-color-success-200);
  border-left-color: var(--eds-color-success-500);
  --dismiss-hover-color: var(--eds-color-success-600);
}

.colorAlert {
  @apply text-alert-700 bg-alert-100;

  /* Left border is thicker and a darker color than the surrounding border */
  border-color: var(--eds-color-alert-200);
  border-left-color: var(--eds-color-alert-500);
  --dismiss-hover-color: var(--eds-color-alert-600);
}
