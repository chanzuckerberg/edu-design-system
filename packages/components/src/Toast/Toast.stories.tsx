import React from "react";
import type { Story } from "@storybook/react";
import Toast from "./Toast";

export default {
  title: "Toast",
  component: Toast,
};

type Args = React.ComponentProps<typeof Toast>;

const Template: Story<Args> = (args: Args) => <Toast {...args} />;

const defaultArgs = {
  children: "You've got toast!",
  onDismiss: () => console.log("dismissed!"),
};

export const Success = Template.bind(null);
Success.args = {
  ...defaultArgs,
  color: "success",
};

export const Alert = Template.bind(null);
Alert.args = {
  ...defaultArgs,
  color: "alert",
};

export const NotDismissable = Template.bind(null);
NotDismissable.args = {
  ...defaultArgs,
  color: "success",
  onDismiss: undefined,
};
