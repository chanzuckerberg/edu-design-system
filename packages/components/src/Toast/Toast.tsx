import CloseButton from "../common/CloseButton";
import NotificationIcon from "../common/Notifications/NotificationIcon";
import React from "react";
import Text from "../Text";
import clsx from "clsx";
import colorStyles from "../common/Notification.module.css";
import styles from "./Toast.module.css";

export type Color = "success" | "alert";

type Props = {
  /**
   * Additional class names passed in for styling.
   */
  className?: string;
  /**
   * The contents of the toast.
   */
  children: React.ReactNode;
  /**
   * The color of the Toast, based on EDS defined colors. Also determines the icon used.
   * Note that the Icon mapping matches the style of Banners.
   */
  color: Color;
  /**
   * Callback when Toast is dismissed.
   */
  onDismiss?: () => void;
};

/**
 * A toast used to provide information on the state of the page, usually in response to a
 * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
 * data was successfully saved.
 */
export default function Toast({
  children,
  className,
  color,
  onDismiss,
  // Allow for additional attributes such as aria roles
  ...rest
}: Props) {
  return (
    <div
      className={clsx(
        className,
        styles.toastDialog,
        color === "success" && colorStyles.colorSuccess,
        color === "alert" && colorStyles.colorAlert,
      )}
      {...rest}
    >
      <div className={styles.content}>
        <NotificationIcon variant={color} />
        <Text color="inherit" size="sm">
          {children}
        </Text>
      </div>
      {onDismiss && (
        <CloseButton color={color} onClose={onDismiss} size="32px" />
      )}
    </div>
  );
}
