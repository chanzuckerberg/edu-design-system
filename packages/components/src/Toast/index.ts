// @flow

export { default } from "./Toast";
