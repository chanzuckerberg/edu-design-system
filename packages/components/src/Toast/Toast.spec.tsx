// @flow

import * as ToastStoryFile from "./Toast.stories";
import { generateSnapshots } from "@chanzuckerberg/story-utils";

describe("<Toast />", () => {
  generateSnapshots(ToastStoryFile);
});
