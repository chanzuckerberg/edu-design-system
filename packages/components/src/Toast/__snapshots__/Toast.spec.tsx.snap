// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> Alert story renders snapshot 1`] = `
<div
  class="toastDialog colorAlert"
>
  <div
    class="content"
  >
    <div
      class="iconAlert"
    >
      <svg
        class="svgIcon"
        fill="currentColor"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          alert
        </title>
        <path
          d="M0,0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M14.9,3H9.1C8.57,3,8.06,3.21,7.68,3.59l-4.1,4.1C3.21,8.06,3,8.57,3,9.1v5.8c0,0.53,0.21,1.04,0.59,1.41l4.1,4.1 C8.06,20.79,8.57,21,9.1,21h5.8c0.53,0,1.04-0.21,1.41-0.59l4.1-4.1C20.79,15.94,21,15.43,21,14.9V9.1c0-0.53-0.21-1.04-0.59-1.41 l-4.1-4.1C15.94,3.21,15.43,3,14.9,3z M15.54,15.54L15.54,15.54c-0.39,0.39-1.02,0.39-1.41,0L12,13.41l-2.12,2.12 c-0.39,0.39-1.02,0.39-1.41,0l0,0c-0.39-0.39-0.39-1.02,0-1.41L10.59,12L8.46,9.88c-0.39-0.39-0.39-1.02,0-1.41l0,0 c0.39-0.39,1.02-0.39,1.41,0L12,10.59l2.12-2.12c0.39-0.39,1.02-0.39,1.41,0l0,0c0.39,0.39,0.39,1.02,0,1.41L13.41,12l2.12,2.12 C15.93,14.51,15.93,15.15,15.54,15.54z"
        />
      </svg>
    </div>
    <p
      class="typography sizeSm colorInherit"
    >
      You've got toast!
    </p>
  </div>
</div>
`;

exports[`<Toast /> Dismissable story renders snapshot 1`] = `
<div
  class="toastDialog colorSuccess"
>
  <div
    class="content"
  >
    <div
      class="iconSuccess"
    >
      <svg
        class="svgIcon"
        fill="currentColor"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          success
        </title>
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
    <p
      class="typography sizeSm colorInherit"
    >
      You've got toast!
    </p>
  </div>
  <button
    class="close button variantLink colorSuccess"
    type="button"
  >
    â€‹
    <svg
      class="svgIcon"
      fill="currentColor"
      height="32px"
      role="img"
      style="--svg-icon-size: 32px;"
      viewBox="0 0 24 24"
      width="32px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</div>
`;

exports[`<Toast /> Success story renders snapshot 1`] = `
<div
  class="toastDialog colorSuccess"
>
  <div
    class="content"
  >
    <div
      class="iconSuccess"
    >
      <svg
        class="svgIcon"
        fill="currentColor"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          success
        </title>
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
    <p
      class="typography sizeSm colorInherit"
    >
      You've got toast!
    </p>
  </div>
</div>
`;
